lib/internal/url.js

```diff
@@ -817,6 +817,9 @@ class URLSearchParams {
     if (arguments.length < 1) {
       throw new TypeError('The `callback` argument needs to be specified');
     }
+    if (typeof callback !== 'function') {
```

Comment:
@cjihrig, this provides more granular error messages, which I think is a good thing. Do you prefer me to just use the same error message for both cases?

In the browser, they have different error messages as well:

```js
>> new URLSearchParams().forEach()
** TypeError: Failed to execute 'forEach' on 'URLSearchParams':
   1 argument required, but only 0 present.
>> new URLSearchParams().forEach(2)
** TypeError: Failed to execute 'forEach' on 'URLSearchParams':
   The callback provided as parameter 1 is not a function.
```